internal/uri/uri_util.go

```diff
@@ -34,7 +34,9 @@ const eclipseWorkspaceFolderScheme = "file:"
 var rangeFragmentRegexp = regexp.MustCompile(`^(.+)://((.*)@)?(.+?)(:(\d*))?/?((.*)\?)?((.*)#)L?(\d+)(?:,(\d+))?(-L?(\d+)(?:,(\d+))?)?`)
 
 func FolderContains(folderPath string, path string) bool {
-	return strings.HasPrefix(path, folderPath)
+	log.Debug().Str("folderPath", folderPath).Str("path", path).Msg("FolderContains")
```

Comment:
This is wrong, I added a failing test to demonstrate

Reply:
Yeah, the folder must be normalized, but the file system path separator included in the contains check. Good catch!

Reply:
Alternatively, we must save all folder paths with a trailing slash, which may be the better solution.

Reply:
Please have a look now - have added a few more test scenarios and adjusted the function

Reply:
Looks OK
I'd probably normalize both paths with filepath.Clean just to make it more robust

Reply:
I'll add that and then merge
=